### PR TITLE
Add `ud30x9` , `sd29x9` , and base functions

### LINKED ISSUE #55: [Feature]: SD59x18 data type
### What is the feature you would like to see?

Implement signed integer up to 59 digits + fractional up to 18 digits data type.

### PR DESCRIPTION
**Pull request by ericnordelo**
Resolves #55 , #56, #57, #58, #59, #60, #61, #62, #64, #65, #67, #68, #69, #71, #73, #91.

#### PR Checklist

- [x] Tests
- [ ] Documentation
- [ ] Changelog

## Summary by CodeRabbit

* **New Features**
  * Introduced UD30x9 fixed-point numeric type with arithmetic operations (addition, subtraction, modulo), bitwise operations (AND, OR, XOR, NOT), and comparison capabilities.
  * Added conversion utilities between standard integers and the fixed-point representation.

* **Tests**
  * Comprehensive test suite for fixed-point operations including edge cases and boundary behavior validation.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: math/fixed_point/Move.toml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+[package]
+name = "openzeppelin_fp_math"
+version = "0.0.1"
```
Comment (bidzyyys) [inline math/fixed_point/Move.toml]: Should we have `version`?

File: math/fixed_point/Move.toml [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+edition = "2024"
+
+[addresses]
+openzeppelin_fp_math = "0x0"
```
[the remark this answers is not in the document]
Comment (ericnordelo) [reply math/fixed_point/Move.toml]: I think the address is still required, to be confirmed tough,

**Review by bidzyyys — APPROVED** [review]
Looks good to me!

File: math/fixed_point/sources/sd29x9/sd29x9_base.move [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+    greater_than_bits(y.unwrap(), x.unwrap())
+}
+
+/// Implements the lower than or equal to operation (<=) for SD29x9 type.
+public fun lte(x: SD29x9, y: SD29x9): bool {
+    !gt(x, y)
+}
+
+/// Implements the checked modulo operation (%) for SD29x9 type.
+public fun mod_(x: SD29x9, y: SD29x9): SD29x9 {
```
Comment (immrsd) [inline math/fixed_point/sources/sd29x9/sd29x9_base.move]: Don't know if it's a type or intentional, seems like we can name it just `mod` here
```suggestion
public fun mod_(x: SD29x9, y: SD29x9): SD29x9 {
```

File: math/fixed_point/sources/sd29x9/sd29x9_base.move [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+
+    if (x_components.neg != y_components.neg) {
+        return !x_components.neg
+    };
+
+    if (!x_components.neg) {
+        x_components.mag > y_components.mag
+    } else {
+        x_components.mag < y_components.mag
+    }
```
Comment (immrsd) [inline math/fixed_point/sources/sd29x9/sd29x9_base.move]:
```suggestion
    if (x_components.neg != y_components.neg) {
        !x_components.neg
    } else if (!x_components.neg) {
        x_components.mag > y_components.mag
    } else {
        x_components.mag < y_components.mag
    }
```

File: math/fixed_point/sources/sd29x9/sd29x9_base.move [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+    wrap_components(result)
+}
+
+/// Implements the AND (&) bitwise operation for SD29x9 type with u128 bits.
+public fun and(x: SD29x9, bits: u128): SD29x9 {
+    from_bits(x.unwrap() & bits)
+}
+
+/// Implements the AND (&) bitwise operation for SD29x9 type with another SD29x9.
+public fun and2(x: SD29x9, y: SD29x9): SD29x9 {
```
[the remark this answers is not in the document]
Comment (ericnordelo) [reply math/fixed_point/sources/sd29x9/sd29x9_base.move]: This is inspired in prb-math, and after some thinking I think I rather use regular `and` for u128 and then `and2` for the other version, reason being that this logical and doesn't really take sign into account, its just a wrapper around bitwise operation in the inner u128 representation. I don't think is better to use `and_u128` since the strong typing already helps users.

File: math/fixed_point/sources/sd29x9/sd29x9_base.move [hunk 6]
```diff
@@ -0,0 +1,232 @@
...
+
+fun wrap_components(value: Components): SD29x9 {
+    if (value.mag == 0) {
+        sd29x9::zero()
+    } else if (value.neg && value.mag == MIN_VALUE) {
+        sd29x9::min()
+    } else {
+        sd29x9::wrap(value.mag, value.neg)
+    }
+}
```
[the remark this answers is not in the document]
Comment (immrsd) [reply math/fixed_point/sources/sd29x9/sd29x9_base.move]: Got it
